app/components/primer/navigation/tab_component.rb

```diff
@@ -97,11 +98,11 @@ def initialize(list: false, selected: false, with_panel: false, icon_classes: ""
         @system_arguments = system_arguments
 
         if with_panel
-          @system_arguments[:tag] ||= :button # rubocop:disable Primer/NoTagMemoize
+          @system_arguments[:tag] = :button
```

Comment:
TODO: verify whether we should allow `button` to be rendered without `with_panel` being set [like done here][0], or whether this linked case should actually be using `with_panel`.


[0]: https://github.com/github/github/blob/master/app/components/discussions/templates/edit_form_component.html.erb#L28

Reply:
Setting `with_panel` will render `tab-container` which comes with it's own keyboard behavior which conflicts with the one specified [here](https://github.com/github/github/blob/master/app/assets/modules/github/behaviors/commenting/preview.ts#L50). This appears to be why `with_panel` is not being used. 

[here]: https://github.com/github/github/blob/master/app/assets/modules/github/behaviors/commenting/preview.ts#L50

Reply:
ANSWER: `with_panel` renders `tab-container` which comes with it's own keyboard behavior which conflicts with the behaviors specified [here](https://github.com/github/github/blob/master/app/assets/modules/github/behaviors/commenting/preview.ts#L50). Therefore, it looks like we want to support tag manually being set to `:button` outside of `with_panel`. 

[here]: https://github.com/github/github/blob/master/app/assets/modules/github/behaviors/commenting/preview.ts#L50

Reply:
I think the "correct" solution would be changing that case to stop relying in custom JS and start using `TabContainer`

Reply:
Agreed! I'll look into how "do-able" that is!